src/foam/nanos/dig/DIG.js

```diff
@@ -133,12 +134,15 @@ foam.CLASS({
           url += "subject=" + subject;
         }
         if ( q ) {
-          url += query ? "&" : "?";
+          //q_ is encoded query part
+          q_ = query ? "&" : "?";
           query = true;
-          url += "q=" + q;
+          q_ += "q=" + encodeURIComponent(q);
```

Comment:
Couldn't we just change this line to:
url += "q=" encodeURIComponent(q);

and keep the rest of the code the same?

Reply:
At the end of the code we are supposed to encode the whole URI. After encoding query part's "&" to %26 then it is encoded again. So I separate and encoded query part from the URl.

Reply:
Have you tested? Create an object with an '&' some field (maybe a script or test) and then see if this works.